datafusion-cli/src/command.rs

```diff
@@ -64,6 +67,11 @@ impl Command {
             Self::Quit => Err(DataFusionError::Execution(
                 "Unexpected quit, this should be handled outside".into(),
             )),
+            Self::ListFunctions => display_all_functions(),
+            Self::SearchFunctions(function) => {
+                let func = function.parse::<Function>().unwrap();
```

Comment:
It seems better to return an err message when the parse fails instead of crashing the program.